File: stockflow-backend/src/main/java/com/stockflow/backend/service/DashboardService.java
package com.stockflow.backend.service;

import com.stockflow.backend.dto.dashboard.*;

import java.util.List;

/**
 * DashboardService — Core interface for executive KPI metrics and visual chart analytics.
 */
public interface DashboardService {

    /** Executive summary KPIs */
    DashboardSummaryResponse getSummary();

    /** Detailed Inventory Stock & Valuation Analytics */
    InventoryAnalyticsResponse getInventoryAnalytics();

    /** Detailed Sales Revenue & Periodic Performance Analytics */
    SalesAnalyticsResponse getSalesAnalytics();

    /** Detailed Purchase Procurement Analytics */
    PurchaseAnalyticsResponse getPurchaseAnalytics();

    /** Monthly Revenue trend data for React charts */
    List<ChartDataResponse> getMonthlyRevenueChart();

    /** Monthly Purchase trend data for React charts */
    List<ChartDataResponse> getMonthlyPurchasesChart();

    /** Category inventory distribution data for React charts */
    List<ChartDataResponse> getCategoryDistributionChart();

    /** Warehouse inventory distribution data for React charts */
    List<ChartDataResponse> getWarehouseDistributionChart();
}

File: stockflow-frontend/src/features/dashboard/services/dashboardService.js
import api from '@/services/api';
import { API } from '@/constants/api';

/**
 * dashboardService — Service functions for dashboard endpoints.
 */
export const dashboardService = {
  getSummary: async () => {
    const res = await api.get(API.DASHBOARD.SUMMARY);
    return res.data;
  },

  getInventoryAnalytics: async () => {
    const res = await api.get(API.DASHBOARD.INVENTORY);
    return res.data;
  },

  getSalesAnalytics: async () => {
    const res = await api.get(API.DASHBOARD.SALES);
    return res.data;
  },

  getPurchaseAnalytics: async () => {
    const res = await api.get(API.DASHBOARD.PURCHASES);
    return res.data;
  },

  getRevenueChart: async () => {
    const res = await api.get(API.DASHBOARD.CHARTS.REVENUE_MONTHLY);
    return res.data;
  },

  getPurchasesChart: async () => {
    const res = await api.get(API.DASHBOARD.CHARTS.PURCHASES_MONTHLY);
    return res.data;
  },

  getCategoryDistribution: async () => {
    const res = await api.get(API.DASHBOARD.CHARTS.CATEGORY_DISTRIBUTION);
    return res.data;
  },

  getWarehouseDistribution: async () => {
    const res = await api.get(API.DASHBOARD.CHARTS.WAREHOUSE_DISTRIBUTION);
    return res.data;
  },
};
